presto-common/src/main/java/com/facebook/presto/common/zorder/ZOrderIntegers.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.zorder;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+public class ZOrderIntegers
+        implements ZOrder
+{
+    public static final int MAX_INPUT_SIZE = 10;
+
+    // Assume mostSignificantSetBitPositions is 1-based
+    private List<Integer> encodingBits;
+    private int totalBitLength;
+    private int maxBitLength;
+
+    public ZOrderIntegers(List<Integer> encodingBits)
+    {
+        this.encodingBits = encodingBits;
+
+        totalBitLength = 0;
+        maxBitLength = 0;
+        for (int bitLength : encodingBits) {
+            totalBitLength += bitLength;
+            maxBitLength = Math.max(maxBitLength, bitLength);
+        }
```

Comment:
See if we can use the following statement to get the totalBitLength and maxBitLength. I know it loops twice, but given the list is short, the cost should be low, but it gives better readability.
`
totalBitLength = encodingBits.stream().mapToInt(Integer::intValue).sum();
maxBitLength = encodingBits.stream().mapToInt(Integer::intValue).max();
`

Reply:
Using the code to find totalBitLength works well; however, the .max() function for maxBitLength actually returns an OptionalInt class, which sometimes does not give a value (java.util.NoSuchElementException: No value present). I can use this code for totalBitLength and loop maxBitLength normally, which will also make the code cleaner.

Reply:
Interesting. If NoSuchElementException could be thrown, then we should handle that accordingly. Do you know in which cases NoSuchElementException could be thrown?

Reply:
Ah, never mind! The error occurred because of a logic error in another part of the code. The code is working now. I'll keep testing to see if the exception is thrown again.

Reply:
It seems the error still exists, although very rarely. I'll try to figure out when this happens.
![image](https://user-images.githubusercontent.com/25407001/122701491-37a10880-d202-11eb-8d03-493f2cdcc2cf.png)